roles/contiv_network/tasks/main.yml

```diff
@@ -1,44 +1,13 @@
 ---
 # This role contains tasks for configuring and starting netmaster and netplugin service
+- name: check dns container image
+  shell: docker inspect skynetservices/skydns
+  register: docker_inspect_result
+  ignore_errors: yes
 
-- name: download netmaster and netplugin
-  get_url:
-    validate_certs: no
-    url: https://github.com/contiv/netplugin/releases/download/v0.0.0-12-11-2015.20-54-40.UTC/netplugin-v0.0.0-12-11-2015.20-54-40.UTC.tar.bz2
-    dest: /tmp/contivnet.tar.bz2
+- name: pull dns container image
+  shell: docker pull skynetservices/skydns
+  when: "'No such image' in docker_inspect_result.stderr"
```

Comment:
On Thu, Dec 31, 2015 at 10:43:17AM -0800, Madhav Puri wrote:
> >  
> > -- name: download netmaster and netplugin
> > -  get_url:
> > -    validate_certs: no
> > -    url: https://github.com/contiv/netplugin/releases/download/v0.0.0-12-11-2015.20-54-40.UTC/netplugin-v0.0.0-12-11-2015.20-54-40.UTC.tar.bz2
> > -    dest: /tmp/contivnet.tar.bz2
> > +- name: pull dns container image
> > +  shell: docker pull skynetservices/skydns
> > +  when: "'No such image' in docker_inspect_result.stderr"
> 
> the pull should be a noop if the image exists, we don't need this conditional, right?

It's a significant, unnecessary network hit and would save a little time for a
little code, so I am +1 for this part of the patch.

-Erik